pkg/trace/pb/appsec.go

```diff
@@ -0,0 +1,39 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package pb
+
+// AppSecRule represents an AppSec rule.
+type AppSecRule struct {
```

Comment:
I wonder if there's an equivalent of `json.RawMessage` in our msgpack library. This would allow to only unserialise the parts we need to obfuscate while keeping the other values in their raw encoding format.

Happy to help digging into this.

Reply:
The msgpack deserialization code is generated from the protobuf schema (with  https://github.com/tinylib/msgp) so there is no partial parsing. We are browsing most of the structure here so not sure it would help much performance-wise.

Reply:
Ok, sounds good.
I was also considering this option as a way to simplify the type definitions by removing the parts that are not visited by the obfuscator.